This introduces NAND bad block table support on the S3C2410, which is required
by the dynamic partition support

Signed-off-by: Harald Welte <laforge@openmoko.org>

Index: u-boot.git/cpu/arm920t/s3c24x0/nand.c
===================================================================
--- u-boot.git.orig/cpu/arm920t/s3c24x0/nand.c	2007-02-06 11:59:05.000000000 +0100
+++ u-boot.git/cpu/arm920t/s3c24x0/nand.c	2007-02-06 12:10:58.000000000 +0100
@@ -188,7 +188,11 @@
 	nand->dev_ready = s3c2410_dev_ready;
 
 	nand->eccmode = NAND_ECC_SOFT;
+#ifdef CONFIG_S3C2410_NAND_BBT
+	nand->options = NAND_USE_FLASH_BBT;
+#else
 	nand->options = 0;
+#endif
 	//nand->waitfunc = dfc_wait;
 
 	//nand->cmdfunc = s3c2410_cmdfunc;
Index: u-boot.git/drivers/nand/nand_bbt.c
===================================================================
--- u-boot.git.orig/drivers/nand/nand_bbt.c	2007-02-06 11:38:01.000000000 +0100
+++ u-boot.git/drivers/nand/nand_bbt.c	2007-02-06 12:10:58.000000000 +0100
@@ -296,6 +296,8 @@
 		for (j = 0; j < len; j++) {
 			if (check_pattern (&buf[j * scanlen], scanlen, mtd->oobblock, bd)) {
 				this->bbt[i >> 3] |= 0x03 << (i & 0x6);
+				printk (KERN_WARNING "Bad eraseblock %d at 0x%08x\n",
+					i >> 1, (unsigned int) from);
 				break;
 			}
 		}
Index: u-boot.git/include/configs/neo1973.h
===================================================================
--- u-boot.git.orig/include/configs/neo1973.h	2007-02-06 12:10:21.000000000 +0100
+++ u-boot.git/include/configs/neo1973.h	2007-02-06 12:10:58.000000000 +0100
@@ -251,4 +251,6 @@
 #define LCD_VIDEO_ADDR		0x33d00000
 #endif
 
+#define CONFIG_S3C2410_NAND_BBT		1
+
 #endif	/* __CONFIG_H */
